### DISCUSSION TITLE
Fine tuning the provided trained HiFiC models

### DISCUSSION THREAD

**Discussion by AliMaleky7997** [tensorflow/compression#121] [Q&A] [answered]
Hi, I am trying to use a trained HiFiC model as a starting point for fine-tuning on a specific dataset of mine. However, I'm not sure if and how I can do this. I understand that the `tfci.py` file contains meta graphs to load your trained models (hific-lo, -mi, and -hi) but I can't seem to understand how I can convert them to a `hific.model.HiFiC` object which is used for training in `train.py`.

I was wondering if you could help me figure out how to do it and if it is not possible, name the dataset on which your provided trained models are trained (or any other specific training script that gives the exact result of your trained models) so I can have the checkpoint of a trained HiFiC model that I can use for fine-tunning.

Many thanks in advance.

**Comment by fab-jul** [comment, marked as answer]
Hi,
Unfortunately, there is no 1-to-1 mapping from the hific.model.HiFiC code to the trained models, because of some unreleased internal layers.
I would try training with the provided trainer, and then finetuning, see https://github.com/tensorflow/compression/tree/master/models/hific#training-your-own-models.

Please let me know if you run into issues, I am happy to help.
Fabian

**Reply by AliMaleky7997** [reply]
Thanks, @fab-jul for your response. I had tried a bit to map the trained models to a trainable hific.model.HiFiC object but didn't succeed. I would definitely try training with the scripts and then fine-tune the learned model on my own data. But one thing I wanted to ask is regarding the following statement in the readme file `Note that it's using coco2014 from tfds which likely needs to be adapted to a bigger dataset to obtain good results`. It makes me wonder if the data on which your provided models are trained is different from the coco2014 dataset or the model is trained on a mixture of different large datasets.

I would be very thankful if you could help me with details that help me get almost the same results as the provided models (or at least the closest it can be as you mentioned that some internal layers have not been released). The provided model is powerful in compressing general scenes but I was trying to test whether it would do better on more specific scenes if I fine-tune it on a narrower set of scenes. That is why it is critical for me to have a pre-trained model which is very close to the provided models.

Thank you so much in advance.

**Reply by fab-jul** [reply]
Sorry for the delay.

> It makes me wonder if the data on which your provided models are trained is different from the coco2014 dataset or the model is trained on a mixture of different large datasets.

We trained on a large internal dataset, however, I think these days we expect the model to perform similarly if you train on coco2014, or maybe the CLIC training set. The note in the README is more a disclaimer, since the paper used the other dataset and we were not able to fully ablade differences.

Let me know if you have specific questions or problems with training.
